### PR TITLE
MINOR: Add missing unit tests for {Local|Remote}LeaderEndpoint classes

### PR DESCRIPTION
**Pull request by kowshik**
I've added unit tests that were previously missing for the `LeaderEndpoint.fetchEpochEndOffsets()` public method.

### CODE REVIEW

**Comment by kowshik** [comment]
Hi @junrao / @satishd / @mattwong949 -- Please could you help review this PR?

File: core/src/test/scala/kafka/server/LocalLeaderEndPointTest.scala [hunk 1]
```diff
@@ -118,6 +120,46 @@ class LocalLeaderEndPointTest {
...
+        .setLeaderEpoch(4)))
+
+    expected = Map(
+      topicPartition -> new EpochEndOffset()
+        .setPartition(topicPartition.partition)
+        .setErrorCode(Errors.NONE.code)
+        .setLeaderEpoch(4)
+        .setEndOffset(6L))
+
+    assertEquals(expected, result)
```
Comment (junrao) [inline core/src/test/scala/kafka/server/LocalLeaderEndPointTest.scala:160]: Could we also test a missing epoch like 3?

**Review by junrao — COMMENTED** [review]
@kowshik : Thanks for the PR. Just one comment.

File: core/src/test/scala/kafka/server/LocalLeaderEndPointTest.scala (hunk 1 above)
Comment (kowshik) [reply core/src/test/scala/kafka/server/LocalLeaderEndPointTest.scala:160]: Done.

**Comment by kowshik** [comment]
@junrao Thanks for the review! I've addressed the comment in bc94d6d3a7a541e5d84d735b45ea4435f63a0974.

**Review by satishd — APPROVED** [review]
Thanks @kowshik for adding the missing test for fetching `EpochEndOffsets` API, LGTM.

File: core/src/test/scala/kafka/server/LocalLeaderEndPointTest.scala [hunk 2]
```diff
@@ -118,6 +120,61 @@ class LocalLeaderEndPointTest {
...
+
+    // Check missing epoch
+    result = endPoint.fetchEpochEndOffsets(Map(
+      topicPartition -> new OffsetForLeaderPartition()
+        .setPartition(topicPartition.partition)
+        .setLeaderEpoch(5)))
```
Comment (junrao) [inline core/src/test/scala/kafka/server/LocalLeaderEndPointTest.scala:181]: Could we add another test for the missing epoch in the middle like 3?

**Review by junrao — COMMENTED** [review]
@kowshik : Thanks for the updated PR. One more comment. Also, are the test failures related?

File: core/src/test/scala/kafka/server/LocalLeaderEndPointTest.scala (hunk 2 above)
Comment (kowshik) [reply core/src/test/scala/kafka/server/LocalLeaderEndPointTest.scala:181]: Done in 8fdf69e.

**Comment by kowshik** [comment]
@junrao Thanks for the review! I've addressed the comment, PR is ready for another pass. The CI test failures look unrelated to this PR.

**Review by junrao — APPROVED** [review]
@kowshik : Thanks for the PR. LGTM

**Pull request merged**